src/pipeline_v3/core/index_manager_atomic.py

```
@@ -169,48 +171,50 @@
-        index_types: IndexType = IndexType.BOTH
+        index_types: IndexType = IndexType.BOTH,
     ) -> bool:
         """
         Update document with atomic guarantees.
-        
+
         Ensures consistent update across all systems or rollback on failure.
         """
         if not self.enable_transactions:
             # Fall back to delete + add
             self.delete_document(doc_id)
```

Pylint (3.3.7)

[error] Instance of 'AtomicIndexManager' has no 'delete_document' member

(E1101)